hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/SyntheticTxnFactory.java

```diff
@@ -210,30 +210,24 @@ public Optional<TransactionBody.Builder> synthContractOpFromEth(final EthTxData
     public TransactionBody.Builder synthContractAutoRemove(final EntityNum contractNum) {
         final var op =
                 ContractDeleteTransactionBody.newBuilder()
+                        .setPermanentRemoval(true)
                         .setContractID(contractNum.toGrpcContractID());
-        return TransactionBody.newBuilder()
-                .setTransactionID(
-                        TransactionID.newBuilder().setAccountID(contractNum.toGrpcAccountId()))
-                .setContractDeleteInstance(op);
+        return TransactionBody.newBuilder().setContractDeleteInstance(op);
     }
 
     public TransactionBody.Builder synthAccountAutoRemove(final EntityNum accountNum) {
         final var grpcId = accountNum.toGrpcAccountId();
         final var op = CryptoDeleteTransactionBody.newBuilder().setDeleteAccountID(grpcId);
-        return TransactionBody.newBuilder()
-                .setTransactionID(TransactionID.newBuilder().setAccountID(grpcId))
-                .setCryptoDelete(op);
+        return TransactionBody.newBuilder().setCryptoDelete(op);
```

Comment:
On line 220 , don't we need to set `.setPermanentRemoval(true)` ?

Reply:
Cleared as account removal is not active and the field is not added to protobufs yet.